Source/Flow/Public/Types/FlowDataPinResults.h

```diff
@@ -0,0 +1,291 @@
+// Copyright https://github.com/MothCocoon/FlowGraph/graphs/contributors
+
+#pragma once
+
+#include "Types/FlowPinEnums.h"
+
+#include "GameplayTagContainer.h"
+
+#include "FlowDataPinResults.generated.h"
+
+USTRUCT(BlueprintType, DisplayName = "Flow DataPin Result")
+struct FFlowDataPinResult
+{
+	GENERATED_BODY()
+
+public:
+
+	// Result for the DataPin resolve attempt
+	UPROPERTY(BlueprintReadWrite, Category = DataPins)
+	EFlowDataPinResolveResult Result = EFlowDataPinResolveResult::FailedUnimplemented;
+
+public:
+	FFlowDataPinResult() { }
+	FFlowDataPinResult(EFlowDataPinResolveResult InResult) : Result(InResult) { }
+};
+
+// Recommend implementing FFlowDataPinResult... for every EFlowPinType
+FLOW_ASSERT_ENUM_MAX(EFlowPinType, 12);
+
+USTRUCT(BlueprintType, DisplayName = "Flow DataPin Result (Bool)")
+struct FFlowDataPinResult_Bool : public FFlowDataPinResult
+{
+	GENERATED_BODY()
+
+public:
+
+	UPROPERTY(BlueprintReadWrite, Category = DataPins)
+	bool Value = false;
+
+public:
+
+	FFlowDataPinResult_Bool() { }
+	FFlowDataPinResult_Bool(bool InValue)
+		: Super(EFlowDataPinResolveResult::Success)
+		, Value(InValue)
+		{ }
+};
+
+USTRUCT(BlueprintType, DisplayName = "Flow DataPin Result (Int)")
+struct FFlowDataPinResult_Int : public FFlowDataPinResult
```

Comment:
I named these results "int" and "float", even though I wavered between int64, double and float64.  

The internals are the largest size (int64, double), but I ultimately ended omitting the 64 from the name to make the structs more readable with the rest of the typed results.